### PR TITLE
Merge Appsembler/eucalyptus/develop into Appsembler/eucalyptus/master

### PR DESCRIPTION
**Pull request by melvinsoft**
This is the final merge for get Azure backups on eucalyptus, new vagrant files by @denibertovic and the new basic server playbook by @johnbaldwin

### CODE REVIEW

**Comment by melvinsoft** [comment]
[(2) Enable backups on Azure for Eucalyptus deployments](https://trello.com/c/Sg1Ynwth/958-2-enable-backups-on-azure-for-eucalyptus-deployments)

File: playbooks/eucalyptus_basic_server.yml [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+      when: COMMON_ENABLE_OAUTH_CLIENT
+    - oraclejdk
+    - role: elasticsearch
+      when: "'localhost' in EDXAPP_ELASTIC_SEARCH_CONFIG|map(attribute='host')"
+    - forum
+    - { role: notifier, NOTIFIER_DIGEST_TASK_INTERVAL: "5" }
+    - { role: "xqueue", update_users: True }
+    - certs
+    - edx_ansible
+    - { role: swapfile, SWAPFILE_SIZE: "2GB" }
```
Comment (tkeemon) [inline playbooks/eucalyptus_basic_server.yml]: We shouldn't need this `swapfile` role. It's being run earlier in the playbook.

Comment (johnbaldwin) [inline playbooks/eucalyptus_basic_server.yml]: good catch. Thx, @tkeemon

**Comment by johnbaldwin** [comment]
No additional comments besides that we might want to make it easier for developers to switch between develop and master branches for Vagrantfile.eucalyptus

**Comment by mrgnr** [comment]
@melvinsoft Which playbook is used for single-server deployments? We need to make sure that it calls the backups role. Otherwise, looks good.

File: playbooks/eucalyptus_basic_server.yml [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+    - role: datadog
+      when: COMMON_ENABLE_DATADOG
+    - role: splunkforwarder
+      when: COMMON_ENABLE_SPLUNKFORWARDER
+    - role: newrelic
+      when: COMMON_ENABLE_NEWRELIC
```
Comment (mrgnr) [inline playbooks/eucalyptus_basic_server.yml]: We don't use Datadog, Splunk, or New Relic, so we can probably skip these roles.

File: playbooks/eucalyptus_basic_server.yml [hunk 3]
```diff
@@ -0,0 +1,77 @@
+---
+# for single server community
+# This file based on `edx_sandbox.yml`
+
+- name: Configure instance(s)
+  hosts: all
+  become: True
```
Comment (mrgnr) [inline playbooks/eucalyptus_basic_server.yml]: You also need to specify `become_method: sudo` (or `become_user: root`).

**Comment by melvinsoft** [comment]
@tkeemon @johnbaldwin @mrgnr @denibertovic Thanks all for review this. Regarding to the changes requested, I'm planning to merge this PR as is, and capture the changes requested in card, so we can apply the changes on `appsembler/eucalyptus/develop` and then merge, because if we change now, we will have new commits on `appsembler/eucalyptus/develop`. Any of the requested changes are bugs. Let me know what you think.

**Comment by bezidejni** [comment]
Just a comment - I remember having a discussion with @denibertovic about Vagrantfiles and branching, I think the conclusion was to have the eucalyptus Vagrantfile in `appsembler/eucalyptus/develop` (same for Dogwood), instead of having multiple Vagrantfiles in the latest branch (Vagrantfile.dogwood, Vagrantfile.eucalyptus). Can we make a decision on that?

**Comment by johnbaldwin** [comment]
@bezidejni I think that will be more maintainable. I'm in favor of it

**Comment by melvinsoft** [comment]
[Configuration changes](https://trello.com/c/iOYeVbqP/979-configuration-changes)

File: playbooks/eucalyptus_basic_server.yml (hunk 3 above)
Comment (melvinsoft) [inline playbooks/eucalyptus_basic_server.yml]: @mrgnr I've found this comment in the `sudo` role:
> # Workaround. What we really want to do is `become_method: sudo su` to run
> # tasks as other users. That's not currently possible in Ansible, but may be in
> # the future: https://github.com/ansible/ansible/issues/12686. Note that
> # modifying /etc/sudoers requires root privileges, so the user running this
> # task must already have the ability to sudo as root.

And also, I noticed that `amc.yml` is using this role in the playbook, maybe is the way to go, any thoughts?

Comment (mrgnr) [inline playbooks/eucalyptus_basic_server.yml]: @melvinsoft Where are you seeing that? I [see it](https://github.com/appsembler/wharf/blob/develop/deploy/roles/common/tasks/main.yml#L41-L52) in the Wharf `common` role.

In Wharf, this was a workaround to allow running a task as another user. For example, I deploy as user `morgan`, but want to run a task as user `nginx`. Ansible doesn't directly support that, so we had to add e.g. `morgan ALL=(ALL) NOPASSWD: ALL` to `/etc/sudoers`.

Comment (melvinsoft) [inline playbooks/eucalyptus_basic_server.yml]: @mrgnr is [here](https://github.com/appsembler/configuration/blob/appsembler/eucalyptus/develop/playbooks/roles/sudo/tasks/main.yml) is a separate role in `configuration`. I think if we include it, we can avoid to have to edit the `sudoers` file on every new deploy.

**Pull request merged**